### PR TITLE
[OSDEV-1000] Move facility processing received from api to a separate class

### PR DESCRIPTION
**Pull request by mazursasha1990**
[OSDEV-1000](https://opensupplyhub.atlassian.net/browse/OSDEV-1000) -  Move facility processing received from API to a separate class `ProcessingFacility`.

The functionality of processing a new facility, which was previously in `facilities_view_set.py`, has been moved to `processing_facility.py.` 
This marks the initial step in implementing a single class responsible for managing the processing of new facilities from both API requests and list uploads.

### CODE REVIEW

**Review by vladsha-dev — CHANGES_REQUESTED** [review]
It seems like there's no consolidation between the lists and the API when it comes to saving facilities. It's simply a matter of moving the facility saving process from the API to a separate function.

**Comment by mazursasha1990** [comment]
> It seems like there's no consolidation between the lists and the API when it comes to saving facilities. It's simply a matter of moving the facility saving process from the API to a separate function.

You are right. I've decided on the following approach for consolidation:

- Move facility creation received from the API to a separate class called `CreateFacility`.
- Move facility creation received from list uploads to another separate class, also named `CreateFacility`.
- Consolidate these two functions as much as possible.

I plan to implement all of these steps in different PRs.

**Review by killev — APPROVED** [review]
LGTM, Approved

File: src/django/api/facility_actions/create_facility.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        result = {
+            'matches': [],
+            'item_id': item.id,
+            'geocoded_geometry': None,
+            'geocoded_address': None,
+            'status': item.status,
+        }
+
+        try:
+            (item, row.fields)
```
Comment (vladsha-dev) [inline src/django/api/facility_actions/create_facility.py:87]: The function call is missing here.

**Review by vladsha-dev — CHANGES_REQUESTED** [review]
1. Left one comment.
2. Please fill in the Release notes file.

**Review by vladsha-dev — APPROVED** [review]
LGTM

File: src/django/api/views/facility/facilities_view_set.py [hunk 2]
```diff
@@ -604,194 +583,19 @@ def create(self, request):
...
+        processed_data = contri_cleaner.process_data()
+        rows = processed_data.rows
+        row = rows[0]
 
-        if (should_create
-                and result['status'] not in errors_status):
-            return Response(result, status=status.HTTP_201_CREATED)
-        else:
-            return Response(result, status=status.HTTP_200_OK)
+        return CreateFacility.createApi(request, row, source, should_create)
```
Comment (roman-stolar) [inline src/django/api/views/facility/facilities_view_set.py:598]: @killev @mazursasha1990 I'm not sure that **CreateFacility** is a good name for it. As there is not always new facility creation happening, there could be "matched", "pending (potential match)". I suggest name it "ProcessingFacility" or something similar.

Comment (vladsha-dev) [reply src/django/api/views/facility/facilities_view_set.py:598]: Good idea

Comment (mazursasha1990) [reply src/django/api/views/facility/facilities_view_set.py:598]: You are right. 
Thanks!

**Review by roman-stolar — APPROVED** [review]

**Pull request merged**